packages/ember-application/lib/system/application-instance.js

```diff
@@ -79,7 +87,9 @@ let ApplicationInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
 
     var application = get(this, 'application');
 
-    set(this, 'rootElement', get(application, 'rootElement'));
+    if (!isEnabled('ember-application-visit')) {
+      set(this, 'rootElement', get(application, 'rootElement'));
```

Comment:
I only did this to preserve the /exact/ behavior without the flag to de-risk (otherwise I could just move this into `boot` too), but I will change the feature flagged version